src/tracer.cc

```diff
@@ -1022,10 +1069,13 @@ TraceManager::TraceFile::SaveTraces(
 }
 
 std::shared_ptr<TraceManager::Trace>
-TraceManager::TraceSetting::SampleTrace()
+TraceManager::TraceSetting::SampleTrace(bool force_sample)
 {
   bool create_trace = false;
-  {
+  // [FIXME: DLIS-6033]
+  // A current WAR for initiating trace based on propagated context only
+  // Currently this is implemented through setting trace rate = 0
+  if (!force_sample && rate_ != 0) {
```

Comment:
So if I'm understanding correctly, for `rate_ > 0` - we will always trace when a context is propagated (client header provided -> force_sample=True) - but these propogation-initiated traces will not affect the counts/logic usedwhen deciding to do normal traces (ex: every 1000th request) ? 

So for example with default `rate == 1000`, before:
- request 0 and request 1000 will be traced. 

But now. if I added client headers to some intermediate request say request 7, then instead:
-  request 0, request 7 (context propagation), and request 1001 would get traced?

If so, will this have any impact or confusing behavior on related tools like trace_summary tool or others?

Reply:
Also out of curiosity, can `rate_ == 0` here? I didn't look through all the code, but I noticed it would probably hit a division by zero / floating point exception here without this check, so I wanted to try it out:
```
tritonserver --model-store /mnt/triton/models --trace-rate 0 --trace-level TIMESTAMPS
```
and I sent some inferences, but didn't get any errors.

Reply:
Yes, that is also what I found. I think I talk about it in the description of this PR. `% by 0` was definitely happening before this PR, and I have no idea why it wasn't segfaulting. With this PR, if `rate_==0`, we are skipping this block.

Reply:
Regarding earlier comment: Good catch. We've discussed with Guan to not update counters, when tracing is happening "by force". Let me correct this.

Reply:
@rmccorm4 and @GuanLuo I've decided to keep updating counters as is (if `rate_` > 0) for now. I've started checking all parameters and it seems like e.g `created_` is not used, so this area would benefit from a clean up.
I made it a scope of this ticket: DLIS-6033